### PR TITLE
Enable Terraform automation on Power, Fix: 33

### PR DESCRIPTION
**Pull request by cs-zhang**
Signed-off-by: CS Zhang <zhangcho@us.ibm.com>

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#" title="Author self-approved">cs-zhang</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **yussufsh** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @yussufsh` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: README.md
Comment (yussufsh) [inline README.md]: We do not have support for Windows yet. I need to work on that issue #66.

Can you please remove it from the commit.

File: docs/quickstart.md
Comment (yussufsh) [inline docs/quickstart.md]: This section should be in https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/docs/automation_host_prereqs.md#automation-host-setup where we ask user to setup Terraform binary.

**Comment by yussufsh** [comment]
@bpradipt @cs-zhang Do you think the shell script is needed here?

IMO we should give the steps to setup providers on ppc64le and users follow all the steps. It is simple download/extract and use it in `terraform init` for Power hosts.

If user wants to use the automation to setup TF and relevant deps then we should point him to use openshift-install-powervs script instead of writing the automation again in this project.

Let me know your thoughts.

**Comment by bpradipt** [comment]
> @bpradipt @cs-zhang Do you think the shell script is needed here?
> 
> IMO we should give the steps to setup providers on ppc64le and users follow all the steps. It is simple download/extract and use it in `terraform init` for Power hosts.
> 
> If user wants to use the automation to setup TF and relevant deps then we should point him to use openshift-install-powervs script instead of writing the automation again in this project.
> 
> Let me know your thoughts.

Yeah, duplicating these setup scripts will create maintenance nightmare going forward. Let's use the existing openshift-install-powervs script for all setup as needed. Let's continue to update it as well.
For this TF project let's just add documentation on how to get the binaries, plugins etc. on Power

**Comment by cs-zhang** [comment]
OK, I'll update the doc to add the steps to download TF and providers.

**Comment by ltccci** [comment]
@cs-zhang: Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.
Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: README.md
Comment (cs-zhang) [reply README.md]: Done

**Review by cs-zhang — COMMENTED** [review]
Done

**Comment by yussufsh** [comment]
The steps should go in https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/docs/automation_host_prereqs.md#automation-host-setup

**Comment by ltccci** [comment]
@cs-zhang: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pull-ocp4-upi-powervm-terraform-validate | e7eb528962523f0112f579d6a1fed8c4e7cc0077 | [link](https://prow.ppc64le-cloud.org/view/s3/prow-logs/pr-logs/pull/ocp-power-automation_ocp4-upi-powervm/166/pull-ocp4-upi-powervm-terraform-validate/1363881458904600576) | `/test pull-ocp4-upi-powervm-terraform-validate`

[Full PR test history](https://prow.ppc64le-cloud.org/pr-history?org=ocp-power-automation&repo=ocp4-upi-powervm&pr=166). [Your PR dashboard](https://prow.ppc64le-cloud.org/pr?query=is:pr+state:open+author:cs-zhang).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by ltccci** [comment]
@cs-zhang: Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.
Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: docs/quickstart.md
Comment (cs-zhang) [reply docs/quickstart.md]: done.

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#pullrequestreview-597250601" title="LGTM">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#" title="Author self-approved">cs-zhang</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)~~ [bpradipt]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: docs/quickstart.md
Comment (yussufsh) [inline docs/quickstart.md]: This file change is needed?

File: docs/automation_host_prereqs.md
Comment (yussufsh) [inline docs/automation_host_prereqs.md]: With `$HOME/.terraformrc` TF will always look for "${PWD}/providers" for any plugins across all the TF modules on that system. Is it what you intend to?

I would always prefer to use `terraform init --plugin-dir ./` which is better way for each TF project/folder.

**Comment by ltccci** [comment]
New changes are detected. LGTM label has been removed.

File: docs/quickstart.md
Comment (cs-zhang) [reply docs/quickstart.md]: Removed.

File: docs/automation_host_prereqs.md
Comment (cs-zhang) [reply docs/automation_host_prereqs.md]: Updated with --plugin-dir.

**Review by sudeeshjohn — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#pullrequestreview-597250601" title="LGTM">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#" title="Author self-approved">cs-zhang</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#pullrequestreview-598637921" title="LGTM">sudeeshjohn</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)~~ [bpradipt]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by yussufsh — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#pullrequestreview-597250601" title="LGTM">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#" title="Author self-approved">cs-zhang</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#pullrequestreview-598637921" title="LGTM">sudeeshjohn</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/166#pullrequestreview-598663475" title="LGTM">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)~~ [bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
